From 729162f4078cb548344afe1f9f2b1bc8ab09dabf Mon Sep 17 00:00:00 2001
From: Stricted <info@stricted.net>
Date: Mon, 8 Oct 2018 11:51:44 +0200
Subject: [PATCH 1/5] power: properly initialize cluster states

this fixes the following error
hardware/samsung/power/power.c:415:51: error: excess elements in array initializer [-Werror]
    char ON[CLUSTER_COUNT][PARAM_MAXLEN]  = {"1", "1"};

when only one cluster is defined

Change-Id: I49f88297afb8d57d8abd309e22fa667134b7e406
---
 power/power.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/power/power.c b/power/power.c
index f9b268c..66e3127 100644
--- a/power/power.c
+++ b/power/power.c
@@ -416,8 +416,15 @@ static void samsung_power_set_interactive(struct power_module *module, int on)
     char button_state[2];
     int rc;
     static bool touchkeys_blocked = false;
-    char ON[CLUSTER_COUNT][PARAM_MAXLEN]  = {"1", "1"};
-    char OFF[CLUSTER_COUNT][PARAM_MAXLEN] = {"0", "0"};
+
+#define INITIALIZE_CLUSTER_STATE(_name, _value) \
+    char _name[CLUSTER_COUNT][PARAM_MAXLEN] = {}; \
+    for (unsigned int i = 0; i < CLUSTER_COUNT; i++) { \
+        strcpy(_name[i], _value); \
+    }
+
+    INITIALIZE_CLUSTER_STATE(ON, "1");
+    INITIALIZE_CLUSTER_STATE(OFF, "0");
 
     ALOGV("power_set_interactive: %d", on);
 
-- 
2.17.1

